tests/base.conf.js

```diff
@@ -30,7 +30,7 @@ class BaseConfig {
         jasmine.getEnv().addReporter(utils.reporter);
         browser.waitForAngularEnabled(false);
 
-        browser.driver.wait(listenForApi, 120 * 1000, 'API took too long to start up');
+        browser.driver.wait(listenForApi(), 120 * 1000, 'API took too long to start up');
```

Comment:
Now I'm worried that the `setupSettings` below will have the same problem. I wonder if it would be better to use `browser.driver.wait` less. What if we...

```js
browser.driver.wait(startApi(), 150 * 1000, 'API took too long to start up');
browser.driver.sleep(10000); // if needed?
return login(browser);
...
const startApi = () => {
  return listenForApi()
    .then(setupSettings)
    .then(utils.setUserContactDoc)
    .then(setupUser);
};
```

That way we're using code we understand more, and bizarro protractor code less.

Reply:
In that code snippet we haven't eliminated `browser.driver.wait` and we lose the granularity of the logging out. If we want to stick to known interfaces, maybe just pass promises into `wait()` and avoid the mysterious interface taking a "function to evaluate as a condition"?

Reply:
The problem with passing promises is if the `setupSettings()` function executes and returns a promise and makes a request to the server before API is running then the promise will immediately fail. That's why I suggested my promise chain solution.

I'm wondering if protractor lets you pass a `function`, or a `Promise` but not a function that returns a promise, which would be quite dumb...

Reply:
You're right.  Will revisit.